Core.Collectors/Collector/CollectorBase.cs

```diff
@@ -63,8 +63,21 @@ public async Task<bool> ProcessAsync(T collectionNode, long maxPageCount)
             while (batchingHttpRequest.HasNext && counter < maxPageCount && !haltCollection)
             {
                 counter++;
-                HttpResponseMessage response = await batchingHttpRequest.NextResponseAsync(this.authentication).ConfigureAwait(false);
+                RequestResult result = await batchingHttpRequest.NextResponseAsync(this.authentication).ConfigureAwait(false);
 
+                if (!result.IsSuccess())
+                {
+                    foreach (CollectionNode allowListCollectionNode in result.allowListStatus.Continuation())
+                    {
+                        if (allowListCollectionNode is T)
+                        {
+                            await this.ProcessAsync((T) allowListCollectionNode).ConfigureAwait(false);
+                        }
```

Comment:
There might be a better way to auto-cast these stuff, but as long as this is working, I am good for letting this in as it is. I will see if I can make generics happy later on :)